Hypermedia.AspNetCore.Siren/HypermediaResourceFilter.cs

```diff
@@ -33,23 +40,27 @@ public void OnResultExecuting(ResultExecutingContext context)
                 .OfType<IPartialResource>()
                 .Select(filter => filter.PartialResource);
 
-            this._builder.WithClaimsPrincipal(context.HttpContext.User);
+            using (var scope = this._factory.CreateScope())
+            {
+                var provider = scope.ServiceProvider as ServiceCollection;
 
-            resource.Configure(this._builder);
+                var builder = scope.ServiceProvider.GetService<IResourceBuilder>();
 
-            foreach (var partialResource in partialResources)
-            {
-                partialResource.Configure(this._builder);
-            }
+                resource.Configure(builder);
 
-            var actualResponse = this._builder.Build();
+                foreach (var partialResource in partialResources)
+                {
+                    partialResource.Configure(builder);
+                }
 
-            objectResult.DeclaredType = actualResponse.GetType();
-            objectResult.Value = actualResponse;
-        }
+                var actualResponse = await builder.BuildAsync();
 
-        public void OnResultExecuted(ResultExecutedContext context)
-        {
+                objectResult.DeclaredType = actualResponse.GetType();
+                objectResult.Value = actualResponse;
+
+            }
```

Comment:
We could extract this into a separate function.

Reply:
To many arguments for the moment, this will only complicate things.